src/Core/Requests/RequestDataConverter.cs

```diff
@@ -22,48 +22,57 @@ public class RequestDataConverter
         /// Converts the given <paramref name="entity"/> to a hubspot data entity.
         /// </summary>
         /// <param name="entity">The entity.</param>
+        /// <param name="preformConversion">This by default performs a data conversion so that the HubSpot "property-value" syntax will be used for serialization. If this parameter is set to false no data conversion is performed (such that a standard object serialization can be performed). </param>
         /// <returns></returns>
-        public dynamic ToHubspotDataEntity(IHubSpotEntity entity)
+        public dynamic ToHubspotDataEntity(IHubSpotEntity entity, bool preformConversion = true)
         {
-            _logger.LogDebug("Convert ToHubspotDataEntity");
-            dynamic mapped = new ExpandoObject();
-
-            mapped.Properties = new List<HubspotDataEntityProp>();
-
-            _logger.LogDebug("Use nameValue mapping?: {0}", entity.IsNameValue);
-
-            var allProps = entity.GetType().GetProperties();
-            _logger.LogDebug("Have {0} props to map", allProps.Length);
-
-            foreach (var prop in allProps)
+            dynamic mapped;
+            if (preformConversion)
```

Comment:
**suggestion**: this diff / code change can be simplified a lot:

```
if(!performConversion) 
{
   return entity;
}

/// ... the rest of the code as it used to be 
```

Reply:
Good suggestion @nover. I shall change that as you suggest.